src/main/scala/scalismo/statisticalmodel/LowRankGaussianProcess.scala

```diff
@@ -164,6 +164,32 @@ class LowRankGaussianProcess[D <: Dim: NDSpace, DO <: Dim: NDSpace](mean: Vector
 }
 
 /**
+ * Convenience class to speedup sampling from a LowRankGaussianProcess obtained by nearest neighbor interpolation of a DiscreteLowRankGaussianProcess
+ *
+ */
+private[scalismo] class NearestNeighbourInterpolatedLowRankGaussianProcess[D <: Dim: NDSpace, DO <: Dim: NDSpace](mean: VectorField[D, DO], klBasis: KLBasis[D, DO], discreteGP: DiscreteLowRankGaussianProcess[D, DO]) extends LowRankGaussianProcess[D, DO](mean, klBasis) {
```

Comment:
I would move this code to the DiscreteLowRankGaussianProcess.scala file, as it is an implementation detail of that class. It should, in my opinion, nowhere else be visible.